docs/src/pages/components/selects/CompareTextFieldSolution.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+import TextField from '@material-ui/core/TextField';
+import FormHelperText from '@material-ui/core/FormHelperText';
+
+const useStyles = makeStyles(theme => ({
+  field: {
+    margin: theme.spacing(1),
+    minWidth: 120,
+  },
+}));
+
+export default function ControlledOpenSelect() {
+  const classes = useStyles();
+  const [age, setAge] = React.useState(10);
+
+  return (
+    <form autoComplete="off">
+      <FormControl className={classes.field}>
+        <InputLabel htmlFor="age-select">Age</InputLabel>
+        <Select
+          value={age}
+          onChange={event => setAge(event.target.value)}
+          inputProps={{
+            name: 'age-select',
+            id: 'age-select',
+          }}
+        >
+          <MenuItem value={10}>Ten</MenuItem>
+          <MenuItem value={20}>Twenty</MenuItem>
+          <MenuItem value={30}>Thirty</MenuItem>
+        </Select>
+        <FormHelperText>Select implementation</FormHelperText>
+      </FormControl>
+
+      <TextField
+        value={age}
+        onChange={event => setAge(event.target.value)}
+        select
+        className={classes.field}
+        id="age-text-field"
+        label="Age"
+        helperText="TextField implementation"
+        SelectProps={
```

Comment:
I would propose the removal of all the blank props from the example. A mention of SelectProps in the demo description could help.

Reply:
I think they are very nice and useful to have in the demo. 

But prettier made it look like a mess. They were one per line as in the original commit and it looked so pretty :). Perhaps we can improve prettier as in codesandbox, where it contracts into single line where possible.